crates/npmrc/src/custom_deserializer.rs

```diff
@@ -125,4 +135,24 @@ mod tests {
         assert_eq!(store_dir, Path::new("/tmp/xdg_data_home/pacquet/store"));
         env::remove_var("XDG_DATA_HOME");
     }
+
+    #[test]
+    fn test_get_absolute_path_with_home_dir_none() {
+        let home_dir = dirs::home_dir().unwrap();
+        let p = get_absolute_path_with_home_dir(None);
+        assert_eq!(home_dir, p)
+    }
+
+    #[test]
+    fn test_get_absolute_path_with_home_dir_prefix() {
+        let home_dir = dirs::home_dir().unwrap();
+        let p = get_absolute_path_with_home_dir(Some("~/deps/pacquet_home".to_string()));
+        assert_eq!(home_dir.join("deps/pacquet_home"), p);
+    }
+
+    #[test]
+    fn test_get_absolute_path_with_home_dir_absolute() {
+        let p = get_absolute_path_with_home_dir(Some("/tmp/pacquet_home".to_string()));
+        assert_eq!(PathBuf::from("/tmp/pacquet_home"), p);
+    }
```

Comment:
Since you are going to remove `get_absolute_path_with_home_dir`, these tests can be replaced by testing `default_store_dir` directly.